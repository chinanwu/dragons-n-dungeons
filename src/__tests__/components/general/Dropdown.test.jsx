import { shallow } from 'enzyme';
import React from 'react';

import {
  Dropdown,
  mapStateToProps,
} from '../../../components/general/Dropdown.jsx';

jest.unmock('../../../components/general/Dropdown.jsx');
jest.unmock('../../../functions/getThemeClassName');

describe('Dropdown component', () => {
  describe('rendering', () => {
    it('renders Dropdown', () => {
      const wrapper = shallow(<Dropdown id="test" options={[]} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders Dropdown when passed options and theme', () => {
      const wrapper = shallow(
        <Dropdown
          id="test"
          theme="dark"
          options={['light', 'dark', 'halloween']}
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    it('toggles theme when dropdown option is chosen', () => {
      const onChange = jest.fn();
      const event = { target: { value: 'one' } };

      const wrapper = shallow(
        <Dropdown
          id="test"
          theme="light"
          options={['ZERO', 'ONE']}
          onChange={onChange}
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();

      wrapper.find('#dropdown__select').simulate('change', event);
      expect(onChange.mock.calls).toEqual([['one']]);
    });
  });

  describe('mapStateToProps', () => {
    it('returned mapped properties', () => {
      const theme = 'light';
      const state = { theme };
      expect(mapStateToProps({ theme: state })).toEqual({ theme: theme });
    });
  });
});
