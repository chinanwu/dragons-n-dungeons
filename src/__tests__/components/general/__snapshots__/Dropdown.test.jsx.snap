// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component behaviour toggles theme when dropdown option is chosen 1`] = `
<div
  className="Dropdown Dropdown--light"
  id="test"
>
  <select
    className="Dropdown__select Dropdown__select--light"
    id="dropdown__select"
    onChange={[Function]}
  >
    <option
      disabled={true}
      hidden={true}
    />
    <option
      className="Dropdown__option Dropdown__option--light"
      id="dropdown__option-0"
      value="zero"
    >
      ZERO
    </option>
    <option
      className="Dropdown__option Dropdown__option--light"
      id="dropdown__option-1"
      value="one"
    >
      ONE
    </option>
  </select>
</div>
`;

exports[`Dropdown component rendering renders Dropdown 1`] = `
<div
  className="Dropdown"
  id="test"
>
  <select
    className="Dropdown__select"
    id="dropdown__select"
    onChange={[Function]}
  >
    <option
      disabled={true}
      hidden={true}
    />
  </select>
</div>
`;

exports[`Dropdown component rendering renders Dropdown when passed options and theme 1`] = `
<div
  className="Dropdown Dropdown--dark"
  id="test"
>
  <select
    className="Dropdown__select Dropdown__select--dark"
    id="dropdown__select"
    onChange={[Function]}
  >
    <option
      disabled={true}
      hidden={true}
    />
    <option
      className="Dropdown__option Dropdown__option--dark"
      id="dropdown__option-0"
      value="light"
    >
      light
    </option>
    <option
      className="Dropdown__option Dropdown__option--dark"
      id="dropdown__option-1"
      value="dark"
    >
      dark
    </option>
    <option
      className="Dropdown__option Dropdown__option--dark"
      id="dropdown__option-2"
      value="halloween"
    >
      halloween
    </option>
  </select>
</div>
`;
